messaging/src/main/java/org/axonframework/eventhandling/async/PropertySequencingPolicy.java

```diff
@@ -0,0 +1,192 @@
+package org.axonframework.eventhandling.async;
+
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+
+import java.util.function.Function;
+
+import org.axonframework.common.property.Property;
+import org.axonframework.common.property.PropertyAccessStrategy;
+import org.axonframework.eventhandling.EventMessage;
+
+/**
+ * A {@link SequencingPolicy} implementation that extracts the sequence identifier from the event message payload based
+ * on a given property or property extractor. If the event message payload is not of a supported type, a fallback 
+ * sequencing policy is used. By default the fallback sequencing policy raises an exception.
+ *
+ * @param <T>
+ *            The type of the supported event payloads.
+ * @param <K>
+ *            The type of the extracted property.
+ *
+ * @author Nils Christian Ehmke
+ * @since 4.5.2
+ */
+@SuppressWarnings( "rawtypes" )
+public final class PropertySequencingPolicy<T, K> implements SequencingPolicy<EventMessage> {
+
+	private final Class<T> payloadClass;
+	private final Function<T, K> propertyExtractor;
+	private final SequencingPolicy<EventMessage> fallbackSequencingPolicy;
+
+	@SuppressWarnings( "unchecked" )
+	private PropertySequencingPolicy( final Builder builder ) {
```

Comment:
Thanks for using the `Builder` @nils-christian! We tend to make the constructors using a `Builder` protected, though. 
This ensures that the framework can still extend on the instance, providing a different `Builder`.

Reply:
I see. I originally did not intend the class for inheritance. I made the constructor protected and removed the final modifier from the class.